### PR TITLE
fixing failure to fetch files with commas

### PR DESCRIPTION
**Pull request by Fl4m3Ph03n1x**
Addressing issue with: https://github.com/v7labs/darwin-py/issues/357

### CODE REVIEW

File: tests/darwin/dataset/remote_dataset_test.py [hunk 1]
```diff
@@ -378,6 +378,28 @@ def it_works(darwin_client: Client, dataset_name: str, dataset_slug: str, team_s
...
+        url = "http://localhost/api/datasets/1/items?page%5Bsize%5D=500"
+        responses.add(
+            responses.POST,
+            url,
+            json=files_content,
+            status=200,
+        )
+
+        actual = remote_dataset.fetch_remote_files({"filenames": ["Video Sep 09, 10 02 59 AM.mov"]})
+        list(actual)
```
Comment (tomvars) [inline tests/darwin/dataset/remote_dataset_test.py:397]: What's the purpose of line 397?

Comment (Fl4m3Ph03n1x) [reply tests/darwin/dataset/remote_dataset_test.py:397]: To consume the generator so I can then use it to make the asserts.

Comment (andreaazzini) [reply tests/darwin/dataset/remote_dataset_test.py:397]: I don't necessarily love the suggestions in the following thread, but perhaps they look a bit nicer?
https://stackoverflow.com/questions/47456631/simpler-way-to-run-a-generator-function-without-caring-about-items

File: tests/darwin/dataset/remote_dataset_test.py [hunk 2]
```diff
@@ -378,6 +378,28 @@ def it_works(darwin_client: Client, dataset_name: str, dataset_slug: str, team_s
...
+        url = "http://localhost/api/datasets/1/items?page%5Bsize%5D=500"
+        responses.add(
+            responses.POST,
+            url,
+            json=files_content,
+            status=200,
+        )
+
+        actual = remote_dataset.fetch_remote_files({"filenames": ["example,with, comma.mp4"]})
+        list(actual)
```
Comment (andreaazzini) [inline tests/darwin/dataset/remote_dataset_test.py:397]: and we don't need `actual`, so we should be fine with this anyway
```suggestion
        list(remote_dataset.fetch_remote_files({"filenames": ["example,with, comma.mp4"]}))
```

**Review by andreaazzini — APPROVED** [review]
Functionally speaking, it works well.

Tested manually
```python
from collections import deque
from darwin import Client
from pathlib import Path

c = Client.from_config(Path("/Users/andrea/.darwin/config.yaml"))
dataset = c.get_remote_dataset("v7-labs/test-comma")

# outputs `deque([])`
deque(dataset.fetch_remote_files({"filenames": "1,2.png"}))

# outputs `deque([DatasetItem(id=337857499, filename='1,2.png', status='processing', archived=False, filesize=None, dataset_id=401344, dataset_slug='n/a', seq=1, current_workflow_id=None, path='/')])`
deque(dataset.fetch_remote_files({"filenames": ["1,2.png"]}))
```

I'd say let's change that test line and merge!

**Comment by andreaazzini** [comment]
@Fl4m3Ph03n1x when it's merged and released, please remember to message that user who raised the issue, and tell him the good news

File: tests/darwin/dataset/remote_dataset_test.py (hunk 1 above)
Comment (Fl4m3Ph03n1x) [reply tests/darwin/dataset/remote_dataset_test.py:397]: I somehow fail to see how defining an additional function that uses an external library to exaust the generator will make it more readable/mantainable:

```
from collections import deque

def exhaust(generator):
    deque(generator, maxlen=0)
```

Other suggestions are OKish:

```def exhaust(generator): all(generator)``` 
(assuming the generator returns `True`)

But I am still not convinced. 

I could do the same:

```def exhaust(generator): list(generator)``` 

Which would work for all generators, irrespective of their return values. 

I am not sold on the idea, please comment here until tomorrow so I know if I should merge or not.

**Pull request merged**